### ISSUE TITLE
CUPS printing works fine, but no printers appear in Google Cloud Print

### ISSUE THREAD

**Issue by timmrogers** [google/cups-connector#214]
I've set up the gcp-cups-connector before (using Linux Mint) - and while it was a struggle for me (mostly with CUPS), it ultimately worked. 

I'm trying again with Debian 8 and no GUI. CUPS is working and happily printing. But while gcp-cups-connector seems to be running happily, no CUPS-shared printers appear for Google Cloud Print.

I am using local-only printing, default init file. systemd to autorun on boot. (Although if I run it manually, I get the same result.) Status from systemd looks fine. /tmp/cups-connector looks fine. No errors, and it says that there are 6 printers. 

But absolutely no printers appear in Google Cloud Print either on Google Chrome on my Mac or on Android. 

Does anyone has a suggestion? What am I missing?

Thank you!

**Comment by jay0lee** [comment]
If you chose local only, no cloud printers will be created / used. The local printers are only accessible to devices on the same LAN as the cups-connector server.

Sounds like you probably want to go back and rerun the init, this time choosing cloud print.

Jay

**Comment by timmrogers** [comment]
Thanks for the reply, Jay. I can certainly try what you suggest. 

But...I used local only when I used this with Linux Mint, and it worked fine. Both my Mac and my Android phone are on the local LAN. Shouldn't they both see the local printer?

My real goal is to provide printing for students on campus using Chromebooks. We have 6 Brother printers, and I'd like the to be able to see all of them. That is what I had done with the Linux Mint install. I'm just trying to move to Debian 8 and a nice dedicated print server.

Thanks!

**Comment by jay0lee** [comment]
Hi Tim,

  Chrome on Mac should be able to see them assuming your network is not blocking p2p connections and mDNS and the device is on the same subnet as the Linux server. Android does not yet support local printing in Cloud Print afaik though I believe @jacobmarble has said it's coming.

Jay

**Comment by jacobmarble** [comment]
Local printing is supported on Android. It's new, so be sure to send bugs to the devs over there.

**Comment by jacobmarble** [comment]
To diagnose mDNS issues, I use "Bonjour Browser" on Android and "Bonjour Browser" on OSX (different apps, same name).

**Comment by timmrogers** [comment]
Thanks to both of you.

Same network, same firewall et al as was working with Linux Mint and
gcp-cups-connector. The issue - whatever it is - must be with either Debian
8 or with my CUPS or gcp-cups-connector configuration.

Standard install of Debian 8. The other services that I am running - tftp,
RADIUS, sSMTP, NTP, and so on - are working fine. server and Mac on the
same subnet.

Might there be something that needs to be enabled in Debian 8 (as opposed
to Ubuntu/Linux Mint) that might be interfering? Or perhaps something in
the cupsd.conf that would allow printing via CUPS but not advertising the
printers to Google Cloud Print?

**Comment by jacobmarble** [comment]
Yes, it's possible that Avahi is configured not to broadcast on the Debian machine. It's also possible that the Debian and Mint machines are using same proxy names or printer names.
- Please confirm that Bonjour Browser shows the printers (look for "privet") from the Mint machine, but not the Debian machine.
- Check the connector log after several minutes. Avahi doesn't return the relevant error right away. All Avahi log entries (including good ones) include the string "Avahi". See [avahi.go](https://github.com/google/cups-connector/blob/master/privet/avahi.go) for more.

**Comment by timmrogers** [comment]
OK, I've confirmed that avahi is running on the Debian server (it is), and
I've installed avahi-discover (perhaps unnecessary - but I saw some chatter
about the need).

Bonjour Browser shows the three printers that I've configured in CUPS under
"_privet._tcp" - which I assume is good!

And now I see two of the three printers on my Android. And...I see all
three printers on a Chromebook! Yipee!

Thanks!

Does it seem reasonable that the issue was avahi-discover?
